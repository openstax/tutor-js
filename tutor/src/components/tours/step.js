import {
    React, PropTypes, observer,
} from 'vendor';
import Spotlight from './spotlight';
import Floater from 'react-floater';
import STEPS from './custom';
import { colors } from 'theme';

@observer
export default class TourStep extends React.Component {

    static propTypes = {
        step: PropTypes.object.isRequired,
        ride: PropTypes.object.isRequired,
    }

    styles = {
        floater: {
            maxWidth: '100%',
        },
        container: {
            padding: 0,
            borderRadius: '4px',
        },
        options: {
            zIndex: 1501,
        },
    };

    classNameStyles = {
        green: {
            arrow: {
                color: colors.secondary,
            },
        },
        greenNoBody: {
            arrow: {
                color: colors.secondary,
                margin: 14,
            },
            container: {
                minHeight: 0,
                padding: 0,
            },
        },
    };


    render() {
        const { step, ride } = this.props;
        const Step = STEPS[step.customComponent || 'Standard'];
        const styles = { ...this.styles, ...this.classNameStyles[step.className] };

        const tip = (
            <Floater
                open={true}
                debug={ride.context.emitDebugInfo}
                autoOpen={true}
                key={step.anchor_id}
                target={step.target}
                styles={styles}
                disableAnimation={true}
                placement={step.placement}
                content={<Step step={step} ride={ride} />}
            />
        );
        if (step.shouldShowSpotlight) {
            return (
                <Spotlight step={step} ride={ride}>
                    {tip}
                </Spotlight>
            );
        }
        return tip;
    }

}
