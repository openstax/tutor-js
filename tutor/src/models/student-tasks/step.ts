import {
    BaseModel, field, observable, computed, array, action, model, modelize,
    NEW_ID, lazyGetter, hydrateModel, getParentOf,
} from 'shared/model';
import S from '../../helpers/string';
import { pick, get, isNil, omit } from 'lodash';
import { extractCnxId } from '../../helpers/content';
import { MediaActions } from '../../flux/media';
import Time from 'shared/model/time';
import urlFor from '../../api';
import {
    ReferenceBookNode, RelatedContent, ChapterSection, Exercise, StudentTask, StudentTaskStepGroup,
} from '../../models';

class TaskStepContent extends BaseModel {
    update(data: any) {
        Object.assign(this, data);
    }
    requiresAnswerId = false;
}

class StudentTaskVideoStep extends TaskStepContent { }
class StudentTaskExternalStep extends TaskStepContent { }
class StudentTaskInteractiveStep extends TaskStepContent { }
class StudentTaskPlaceHolderStep extends TaskStepContent { }

class StudentTaskReadingStep extends TaskStepContent {
    title = '';
    content_url = ''
    html = ''
    chapter_section = ''

    constructor() {
        super()
        modelize(this)
    }

    @model(RelatedContent) related_content = array<RelatedContent>()

    @lazyGetter get chapterSection() { return new ChapterSection(this.chapter_section) }
    @lazyGetter get page() {
        return hydrateModel(ReferenceBookNode, {
            uuid: this.related_content[0].uuid,
            id: this.related_content[0].page_id,
            cnx_id: extractCnxId(this.content_url),
            content_html: this.html,
            title: this.related_content[0].title,
            chapter_section: this.chapterSection,
        })
    }

    @computed get pageTitle(): string {
        return this.title || get(this, 'related_content[0].title');
    }
}

export
class StudentTaskExerciseStep extends Exercise {
    constructor() {
        super()
        modelize(this)
    }
    get stem_html() { return this.content.stem_html; }
    get questions() { return this.content.questions; }
    get stimulus_html() { return this.content.stimulus_html; }
    get requiresAnswerId() { return this.content.isMultiChoice; }
}

const ContentClasses = {
    video: StudentTaskVideoStep,
    reading: StudentTaskReadingStep,
    exercise: StudentTaskExerciseStep,
    external_url: StudentTaskExternalStep,
    placeholder: StudentTaskPlaceHolderStep,
    interactive: StudentTaskInteractiveStep,
};

const UNSAVEABLE_TYPES = [
    'placeholder',
];

const NO_ADDITIONAL_CONTENT = [
    'external_url',
];

export class StudentTaskStep extends BaseModel {

    @field id = NEW_ID;
    @field uid = '';
    @field preview = '';
    @field available_points = 0;
    @field type = '';
    @field is_completed = false
    @field answer_id? = NEW_ID;
    @field free_response = '';
    @field feedback_html = '';
    @field correct_answer_id = NEW_ID;
    @model(Time) last_completed_at = Time.unknown
    @field response_validation?: any = {}
    @field spy?: any = {}
    @field external_url = ''
    @field labels?: any[];
    @field formats?: any[];
    @field group = ''
    @field can_be_updated = false
    @field dropped_method = ''
    @field is_feedback_available = false
    @field published_points = 0
    @field published_comments = ''
    @field published_points_without_lateness = 0
    @field published_late_work_point_penalty = 0
    @field tasked_id = NEW_ID
    @field exercise_id = NEW_ID

    @observable content?: any
    @observable isFetched = false

    @observable multiPartGroup?: StudentTaskStepGroup

    constructor() {
        super()
        modelize(this)
    }

    get task() { return getParentOf<StudentTask>(this) }

    @computed get canAnnotate() {
        return this.isReading;
    }

    @computed get chapterSection() {
        return this.content.related_content ?
            this.content.related_content[0].chapter_section : null;
    }

    @computed get isExercise() {
        return 'exercise' === this.type;
    }
    @computed get isReading() {
        return 'reading' === this.type;
    }
    @computed get isExternalUrl() {
        return 'external_url' === this.type;
    }
    @computed get isInteractive() {
        return 'interactive' === this.type;
    }
    @computed get isVideo() {
        return 'video' === this.type;
    }
    @computed get isPlaceHolder() {
        return 'placeholder' === this.type;
    }
    @computed get isDroppedQuestion() {
        return Boolean(this.dropped_method);
    }

    // read properties from content when it may not have been fetched yet
    readContentProperty(property: string) {
        if (!this.content) {
            return null;
        }
        return this.content[property];
    }

    @computed get isOpenEndedExercise() {
        return Boolean(this.isExercise && this.readContentProperty('isOpenEnded'));
    }

    @computed get isWrittenResponseExercise() {
        return Boolean(this.isExercise && this.readContentProperty('isWrittenResponse'));
    }

    @computed get isCorrect() {
        return Boolean(
            this.correct_answer_id && this.answer_id == this.correct_answer_id
        );
    }

    @computed get pointsScored() {
        if(!isNil(this.published_points)) return this.published_points;
        if (this.correct_answer_id) {
            return this.answer_id === this.correct_answer_id
                ? this.available_points
                : 0;
        }
        return null;
    }

    @computed get isTwoStep() {
        return Boolean(
            this.isExercise && this.readContentProperty('isMultiChoice') && this.formats?.includes('free-response'),
        )
    }

    @computed get isReview() {
        return this.labels?.includes('review');
    }

    @computed get isPersonalized() {
        return 'personalized' == this.group ;
    }

    @computed get isSpacedPractice() {
        return 'spaced practice' == this.group ;
    }

    @computed get canEditFreeResponse() {
        return Boolean(
            this.can_be_updated && !this.answer_id &&
                this.formats?.includes('free-response') &&
                (this.readContentProperty('isOpenEnded') || S.isEmpty(this.free_response))
        );
    }

    @computed get hasBeenAnswered() {
        return Boolean(this.answer_id);
    }

    @computed get canAnswer() {
        return Boolean(
            this.isExercise && this.can_be_updated
        );
    }

    @computed get needsFetched() {
        return Boolean(
            !NO_ADDITIONAL_CONTENT.includes(this.type) && !this.api.hasBeenFetched
        );
    }

    @computed get isLate() {
        return this.last_completed_at.isAfter(this.task.due_at)
    }

    @action fetchIfNeeded() {
        if (this.needsFetched && !this.api.isInProgress('fetchStudentTaskStep')) {
            this.fetch();
        }
    }

    @action markViewed() {
        if (this.can_be_updated && !this.isExercise && !this.isExternalUrl && !this.is_completed) {
            this.is_completed = true;
            this.save();
        }
    }

    // called when the task is reloaded and each step is reset
    @action setFromTaskFetch(data: any) {
        // the step is being re-used and it's type changed
        if (data.id != this.id || data.type != this.type) {
            this.content = null
            this.isFetched = false
        }
        this.api.reset();
        this.update(data);
    }

    // called by API
    async fetch(): Promise<void> {
        const data = await this.api.request(urlFor('fetchStudentTaskStep', { stepId: this.id }))
        this.onLoaded(data)
    }

    async save(): Promise<void> {
        if (UNSAVEABLE_TYPES.includes(this.type)) { return; }
        const data = await this.api.request(
            urlFor('saveStudentTaskStep', { stepId: this.id }),
            { data: pick(this, 'is_completed', 'answer_id', 'free_response', 'response_validation') }
        )
        this.onLoaded(data)
    }

    @action beginRecordingAnswer({ free_response }: { free_response: string }) {
        this.free_response = free_response;
        if (this.content.requiresAnswerId) {
            this.answer_id = undefined;
        } else {
            this.is_completed = true;
        }
    }

    @action onLoaded(data: any) {
        this.update(omit(data, 'id'))
        const Klass = ContentClasses[this.type];
        if (!Klass) {
            throw new Error(`Attempted to set content on unknown step type ${this.type}`);
        }
        this.content = hydrateModel(Klass, data, this)
        if (this.isReading) {
            MediaActions.parse(this.content.html);
        }
        this.isFetched = true;
    }

}
