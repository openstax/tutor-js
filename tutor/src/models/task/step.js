import {
  BaseModel, identifiedBy, identifier, hasMany, field, belongsTo,
} from '../base';
import { computed } from 'mobx';
import { last } from 'lodash';

import RelatedContent from './step-related-content';

@identifiedBy('task/step')
export default class TaskStep extends BaseModel {

  @identifier id;
  @field task_id;
  @field title;
  @field type;

  @field({ type: 'array' }) chapter_section;
  @field content_html;
  @field content_url;
  @field({ type: 'date' }) first_completed_at;
  @field({ type: 'date' }) last_completed_at;

  @field group;
  @field has_recovery;

  @field is_completed;

  @field({ type: 'array' }) labels;

  @hasMany({ model: RelatedContent }) related_content;

  @belongsTo({ model: 'task' }) task;

  @computed get cnxId() {
    return last( (this.content_url || '').split('contents/') );
  }
}
