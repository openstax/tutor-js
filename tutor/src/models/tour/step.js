import {
  BaseModel, identifiedBy, identifier, belongsTo, field,
} from 'shared/model';

import {
  computed, action,
} from 'mobx';

import { isEmpty, intersection } from 'lodash';

import Markdown from 'markdown-it';
import MDRegex from 'markdown-it-regexp';

import Actions from './actions';
import WindowSize from '../window-size';

const MD = Markdown({ html: true, linkify: true, typographer: true });

MD.use(MDRegex(/:best-practices:/, () => '<i class="tour-step-best-practices"></i>' ));

// TourStep
// A step in a tour, where steps are connected by a “next” button that leads from one step to the next.
// Can be linked to either an anchor or region
// Has a title and rich text body.

@identifiedBy('tour/step')
export default class TourStep extends BaseModel {
  @identifier id;

  @belongsTo tour;

  @field title;
  @field body;
  @field position;
  @field isCancelable = true;
  @field is_fixed;
  @field anchor_id;
  @field customComponent;
  @field spotlight = true;
  @field displayAs = 'standard';
  @field spotLightPadding = 5;
  @field requiredViewsCount = 1;
  @field displayWithButtons = true;
  @field({ type: 'object' }) action;
  @field className;
  @field({ type: 'array' }) disabledBreakpoints;

  windowSize = new WindowSize();

  @computed get target() {
    return this.anchor_id ? `[data-tour-anchor-id="${this.anchor_id}"]` : null;
  }

  get element() {
    return this.target ? document.querySelector(this.target) : null;
  }

  @computed get placement() {
    if (!this.element) { return 'center'; }
    return (this.anchor_id && this.position) ? this.position : 'auto';
  }

  @computed get isCentered() {
    return this.position == 'center';
  }

  @computed get actionClass() {
    if (!this.action) { return null; }
    return Actions.forIdentifier(this.action.id);
  }

  @computed get actionInstance() {
    return this.actionClass && new this.actionClass({
      step: this,
      ...this.action,
    });
  }

  @computed get shouldShowSpotlight() {
    return Boolean((this.anchor_id || this.isCentered) && this.spotlight);
  }

  @action preValidate() {
    this.actionInstance && this.actionInstance.preValidate();
  }

  @action prepare(options) {
    return (this.actionInstance && this.actionInstance.beforeStep(options)) || Promise.resolve();
  }

  @action complete(options) {
    return (this.actionInstance && this.actionInstance.afterStep(options) ) || Promise.resolve();
  }

  get isViewable() {
    if (!isEmpty(intersection(this.disabledBreakpoints, [this.windowSize.currentBreakpoint]))) {
      return false;
    }
    return Boolean(!this.target || this.element);
  }

  @computed get HTML() {
    return this.body ? MD.render(this.body) : '';
  }

  @computed get shouldReplay() {
    return this.requiredViewsCount > this.tour.viewCounts;
  }

}
