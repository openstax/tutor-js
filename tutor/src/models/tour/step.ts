import { BaseModel, field, modelize, computed, action, NEW_ID, getParentOf } from 'shared/model';
import { isEmpty, intersection } from 'lodash';
import Markdown from 'markdown-it';
import MDRegex from 'markdown-it-regexp';
import Actions from './actions';
import WindowSize from '../window-size';
import type Tour from '../tour'

const MD = Markdown({ html: true, linkify: true, typographer: true });
MD.use(MDRegex(/:best-practices:/, () => '<i class="tour-step-best-practices"></i>' ))

// TourStep
// A step in a tour, where steps are connected by a “next” button that leads from one step to the next.
// Can be linked to either an anchor or region
// Has a title and rich text body.

export default class TourStep extends BaseModel {
    @field id = NEW_ID;

    @field title = '';
    @field body = '';
    @field position = '';
    @field isCancelable = true;
    @field is_fixed = false;
    @field anchor_id: string | null = null;
    @field customComponent: any;
    @field spotlight = true;
    @field displayAs = 'standard';
    @field spotLightPadding = 5;
    @field requiredViewsCount = 1;
    @field displayWithButtons = true;
    @field action?: any;
    @field className?: string;
    @field disabledBreakpoints?: any[];

    windowSize = new WindowSize();

    constructor() {
        super();
        modelize(this);
    }

    get tour() { return getParentOf<Tour>(this) }

    @computed get target() {
        return this.anchor_id ? `[data-tour-anchor-id="${this.anchor_id}"]` : null;
    }

    get element(): HTMLElement | null {
        return this.target ? document.querySelector(this.target) : null;
    }

    @computed get placement() {
        if (!this.element) { return 'center'; }
        return (this.anchor_id && this.position) ? this.position : 'auto';
    }

    @computed get isCentered() {
        return this.position == 'center';
    }

    @computed get actionClass() {
        if (!this.action) { return null; }
        return Actions.forIdentifier(this.action.id);
    }

    @computed get actionInstance() {
        return this.actionClass && new this.actionClass({
            step: this,
            ...this.action,
        });
    }

    @computed get shouldShowSpotlight() {
        return Boolean((this.anchor_id || this.isCentered) && this.spotlight);
    }

    @action preValidate() {
        this.actionInstance && this.actionInstance.preValidate();
    }

    @action prepare(options: any) {
        return (this.actionInstance && this.actionInstance.beforeStep(options)) || Promise.resolve();
    }

    @action complete(options: any) {
        return (this.actionInstance && this.actionInstance.afterStep(options) ) || Promise.resolve();
    }

    get isViewable() {
        if (!isEmpty(intersection(this.disabledBreakpoints, [this.windowSize.currentBreakpoint]))) {
            return false;
        }
        return Boolean(!this.target || this.element);
    }

    @computed get HTML() {
        return this.body ? MD.render(this.body) : '';
    }

    @computed get shouldReplay() {
        return this.requiredViewsCount > this.tour.viewCounts;
    }
}
