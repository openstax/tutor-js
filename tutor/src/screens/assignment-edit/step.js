import { React, PropTypes, observer, observable, computed, action } from 'vendor';
import details from './details';
import chapters from './chapters';
import questions from './questions';
import points from './points';
import reorder from './reorder';

import UnknownType from './unknown';

const reading = {
  details,
  chapters,
  reorder,
};

const homework = {
  details,
  chapters,
  questions,
  points,
};

const external = {
  details,
};

const event = {
  details,
};

const StepTypes = {
  homework, reading, external, event,
};


const Step = observer(({ ux }) => {
  const Component = ux.steps.currentStepComponent || UnknownType;
  return <Component ux={ux} />;
});

Step.propTypes = {
  ux: PropTypes.object.isRequired,
};

class StepUX {

  constructor(ux) {
    this.ux = ux;
  }

  @observable _stepIndex = 0;

  @computed get number() {
    return this._stepIndex + 1;
  }

  @computed get headerText() {
    return this.hasSteps ? `STEP ${this.number}` : null;
  }

  @computed get hasSteps() {
    return this.stepIds.length > 1;
  }

  @action.bound async goForward() {
    if (this.canGoForward) {
      // TODO, skip steps if the assignment type doesn't need the next,
      // for instance events won't have chapters & questiosn
      await this.ux.saveAsDraft();

      this.ux.navigateToStep(this.stepIds[this._stepIndex + 1]);
    }
  }

  @action setIndex(index) {
    const stepIndex = this.stepIds.indexOf(index);
    if (-1 !== stepIndex) {
      this._stepIndex = stepIndex;
    }
  }

  @action.bound goBackward() {
    if (this.canGoBackward) {
      this.ux.navigateToStep(this.stepIds[this._stepIndex - 1]);
    }
  }

  @computed get isLast() {
    return this._stepIndex >= this.stepIds.length - 1;
  }

  @computed get isFirst() {
    return this._stepIndex <= 0;
  }

  @computed get stepComponents() {
    return StepTypes[this.ux.plan.type];
  }

  @computed get stepIds() {
    return Object.keys(this.stepComponents || {});
  }

  @computed get currentStepComponent() {
    return this.stepComponents[this.currentStepId];
  }

  @computed get currentStepId() {
    return this.stepIds[this._stepIndex];
  }

  @computed get canGoForward() {
    return Boolean(
      !this.isApiPending &&
        !this.isLast &&
        this.ux.validations.isValid
    );
  }

  @computed get canGoBackward() {
    return Boolean(
      !this.isApiPending && !this.isFirst
    );
  }

  @computed get canSubmit() {
    return Boolean(
      this.ux.validations.isValid
    );
  }

}


export { StepUX, Step, StepTypes };
