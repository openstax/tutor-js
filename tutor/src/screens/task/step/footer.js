import {
    React, PropTypes, observer, styled,
} from 'vendor';
import Theme from '../../../theme';
import { StudentTaskStep, Course } from '../../../models'
import { ExerciseIdentifierLink } from 'shared';
import RelatedContentLink from '../../../components/related-content-link';

const StyledExerciseFooter = styled.div`
  display: flex;
  margin-top: 3rem;
  padding-top: 1rem;
  font-size: 1.4rem;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid ${Theme.colors.neutral.light};

  ${({ theme }) => theme.breakpoint.only.mobile`
    .exercise-identifier-link {
      display: none;
    }
  `}
`;

const StyledExerciseIdentifierLink = styled(ExerciseIdentifierLink)`
color: ${Theme.colors.neutral.lite};
`;


const StepFooter = observer(({ course, step, hideContentLink }) => {
    return (
        <StyledExerciseFooter>
            {!hideContentLink &&
        <RelatedContentLink
            course={course}
            content={step.content.related_content}
        />}
            <StyledExerciseIdentifierLink
                exerciseId={step.uid}
                bookUUID={course.ecosystem_book_uuid}
                related_content={step.content.related_content}
            />
        </StyledExerciseFooter>
    );
});

StepFooter.propTypes = {
    course: PropTypes.instanceOf(Course).isRequired,
    step: PropTypes.instanceOf(StudentTaskStep).isRequired,
    hideContentLink: PropTypes.bool,
};
StepFooter.displayName = 'StepFooter';

export { StepFooter };
