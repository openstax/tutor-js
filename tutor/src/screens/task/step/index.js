import {
  React, PropTypes, observer, styled,
} from 'vendor';
import {
  Facebook as ReadingLoader,
  BulletList as HomeworkLoader,
} from 'react-content-loader';
import Reading from './reading';
import Exercise from './exercise';
import Placeholder from './placeholder';
import HtmlContent from './html-content';
import Failure from './failure';
import End from './end';
import { LoadingCard } from './card';
import {
  PersonalizedGroup,
  TwoStepValueProp,
  IndividualReview,
  SpacedPractice,
} from './value-props';


const STEP_TYPES = {
  end: End,
  reading: Reading,
  video: HtmlContent,
  exercise: Exercise,
  placeholder: Placeholder,
  interactive: HtmlContent,
  'two-step-intro': TwoStepValueProp,
  'personalized-intro': PersonalizedGroup,
  'spaced-practice-intro': SpacedPractice,
  'individual-review-intro': IndividualReview,
};

const PENDING_TYPES = {
  reading: ReadingLoader,
  exercise: HomeworkLoader,
  video: HtmlContent.Loader,
  interactive: HtmlContent.Loader,
};

// has plenty of space at bottom so last step can
// be scrolled up and user knows it's last
const MultipartGroup = styled.div`
  padding-bottom: 200px;
`;

@observer
class TaskStep extends React.Component {

  static propTypes = {
    pending: PropTypes.func,
    ux: PropTypes.object.isRequired,
    step: PropTypes.shape({
      steps: PropTypes.array,
      type: PropTypes.string.isRequired,
      needsFetched: PropTypes.bool,
      api: PropTypes.shape({
        hasErrors: PropTypes.bool,
      }),
    }).isRequired,
  };

  render() {
    const { ux, step } = this.props;

    if (!step || (step.api && step.api.hasErrors)) {
      return <Failure task={ux.task} step={step} />;
    }
    const { type, needsFetched } = step;

    const stepProps = {
      ...this.props,
      onContinue: ux.canGoForward ? ux.goForward : null,
    };

    if ('mpq' === type) {
      return (
        <MultipartGroup>
          {step.steps.map((s, i) =>
            <TaskStep
              key={i}
              {...stepProps}
              isMultiPart
              isFollowupMPQ={0 !== i}
              step={s}
            />)}
        </MultipartGroup>
      );
    }

    if (needsFetched) {
      const Pending = PENDING_TYPES[type] || PENDING_TYPES.reading;
      return <LoadingCard><Pending /></LoadingCard>;
    }

    const Step = STEP_TYPES[type];

    if (!Step) {
      return <Failure task={ux.task} step={step} />;
    }

    return (
      <Step {...stepProps} />
    );
  }
}

export { TaskStep };
