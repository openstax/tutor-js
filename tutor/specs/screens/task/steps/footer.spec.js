import { StepFooter } from '../../../../src/screens/task/step/footer';
import { Factory, TimeMock, C } from '../../../helpers';

describe('Task Step Footer', () => {
    let props;
    TimeMock.setTo('2017-10-14T12:00:00.000Z');

    beforeEach(() => {
        const task = Factory.studentTask({ type: 'homework', stepCount: 1 });
        props = {
            course: Factory.course(), step: task.steps[0],
        };
    });

    it('matches snapshot', () => {
        const exf = mount(<C><StepFooter {...props} /></C>);
        expect(exf.debug()).toMatchSnapshot();
        exf.unmount();
    });

});
