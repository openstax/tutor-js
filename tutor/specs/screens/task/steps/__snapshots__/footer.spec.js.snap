// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Step Footer matches snapshot 1`] = `
"<StepFooter course={{...}} step={{...}}>
  <styled.div>
    <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
      <div className=\\"sc-ifAKCX KjPkQ\\">
        <Styled(ExerciseIdentifierLink) exerciseId=\\"1@1\\" bookUUID=\\"405335a3-7cff-4df2-a9ad-29062a4af261\\" related_content={{...}}>
          <StyledComponent exerciseId=\\"1@1\\" bookUUID=\\"405335a3-7cff-4df2-a9ad-29062a4af261\\" related_content={{...}} forwardedComponent={{...}} forwardedRef={{...}}>
            <ExerciseIdentifierLink exerciseId=\\"1@1\\" bookUUID=\\"405335a3-7cff-4df2-a9ad-29062a4af261\\" related_content={{...}} className=\\"sc-EHOje frzuNi\\">
              <div className=\\"exercise-identifier-link sc-EHOje frzuNi\\">
                ID# 
                1@1
                 | 
                <a target=\\"_blank\\" tabIndex={-1} href=\\"https://openstax.org/errata/form?source=tutor&location=1%401%201.2%20Angiosperms&book=College%20Physics\\">
                  Suggest a correction
                </a>
              </div>
            </ExerciseIdentifierLink>
          </StyledComponent>
        </Styled(ExerciseIdentifierLink)>
        <RelatedContentLink course={{...}} content={{...}} preamble=\\"Comes from\\">
          <styled.div className=\\"related-content-link\\">
            <StyledComponent className=\\"related-content-link\\" forwardedComponent={{...}} forwardedRef={{...}}>
              <div className=\\"related-content-link sc-bwzfXH fVSdij\\">
                <styled.span>
                  <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                    <span className=\\"sc-bxivhb preamble eSqMCu\\">
                      Comes from
                    </span>
                  </StyledComponent>
                </styled.span>
                <BrowseTheBook unstyled={true} chapterSection={{...}} course={{...}} tabIndex={-1} tag=\\"a\\" windowImpl={{...}}>
                  <a aria-label=\\"Browse course reference book\\" onClick={[Function: res]} tabIndex={-1} href=\\"/book/1/section/1.2\\" target=\\"_blank\\" className=\\"browse-the-book\\">
                    <ChapterSection chapterSection={{...}}>
                      <styled.span className=\\"chapter-section\\" data-chapter-section=\\"1.2\\">
                        <StyledComponent className=\\"chapter-section\\" data-chapter-section=\\"1.2\\" forwardedComponent={{...}} forwardedRef={{...}}>
                          <span className=\\"chapter-section sc-bdVaJa hKLgvF\\" data-chapter-section=\\"1.2\\">
                            1.2
                          </span>
                        </StyledComponent>
                      </styled.span>
                    </ChapterSection>
                    <styled.span>
                      <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                        <span className=\\"sc-htpNat title bNgkkP\\">
                          Angiosperms
                        </span>
                      </StyledComponent>
                    </styled.span>
                  </a>
                </BrowseTheBook>
              </div>
            </StyledComponent>
          </styled.div>
        </RelatedContentLink>
      </div>
    </StyledComponent>
  </styled.div>
</StepFooter>"
`;
